analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Extensions/IOperationExtensions.cs

```diff
@@ -102,6 +103,38 @@ public static IOperation UnwrapConversion(this IOperation operation)
             return operation;
         }
 
+        /// <inheritdoc cref="ArgumentValue(ImmutableArray{IOperation}, string)"/>
+        public static IOperation ArgumentValue(this IInvocationOperationWrapper invocation, string parameterName) =>
+            invocation.Arguments.ArgumentValue(parameterName);
+
+        /// <inheritdoc cref="ArgumentValue(ImmutableArray{IOperation}, string)"/>
+        public static IOperation ArgumentValue(this IObjectCreationOperationWrapper objectCreation, string parameterName) =>
+            objectCreation.Arguments.ArgumentValue(parameterName);
+
+        /// <inheritdoc cref="ArgumentValue(ImmutableArray{IOperation}, string)"/>
+        public static IOperation ArgumentValue(this IPropertyReferenceOperationWrapper propertyReference, string parameterName) =>
+            propertyReference.Arguments.ArgumentValue(parameterName);
+
+        /// <inheritdoc cref="ArgumentValue(ImmutableArray{IOperation}, string)"/>
+        public static IOperation ArgumentValue(this IRaiseEventOperationWrapper raiseEvent, string parameterName) =>
+            raiseEvent.Arguments.ArgumentValue(parameterName);
+
+        /// <summary>
+        /// Returns the argument value corresponding to <paramref name="parameterName"/>. For <see langword="params"/> parameter an IArrayCreationOperation is returned.
+        /// </summary>
+        private static IOperation ArgumentValue(this ImmutableArray<IOperation> arguments, string parameterName)
```

Comment:
Even when it's private, I wouldn't make it an extension.
```suggestion
        private static IOperation ArgumentValue(ImmutableArray<IOperation> arguments, string parameterName)
```